connect-examples/v2/graphql-sample-app/index.js

```diff
@@ -72,18 +72,36 @@ function getQueryVariables(query) {
 async function query(queryName, queryVariables) {
   const sampleQuery = QUERIES[queryName];
   try {
+    let hasNextPage = true;
+    let prevEndCursor = null;
+    let pageNum = 1;
     console.log(`Running the query: ${sampleQuery}`);
     console.log(`Here are the results from the ${queryName} query:`);
-    const data = await getGraphQlClient().request(sampleQuery, queryVariables);
-    const dataObj = queryName === "catalog" ? data[queryName].all : data[queryName];
-    if (!dataObj) {
-      console.log("No objects returned from query");
-      return;
-    }
-    const nodes = dataObj.nodes;
-    for (const obj of nodes) {
-      console.log("===================================================");
-      console.dir(obj, { depth: null });
+    while (hasNextPage) {
+      if (prevEndCursor) {
```

Comment:
Could collapse this conditional into a 1 liner with [nullish coalescing](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_operator):
```js
queryVariables.after = prevEndCursor ?? null;
```
If we want to avoid newer ECMAscript features, it could just be a ternary too.

Or if we are sure that prevEndCursor is only ever a valid cursor or `null`, we could assign it directly to queryVariables.after without any conditional wrapper:
```js
queryVariables.after = prevEndCursor;
```